Not Documented Error Responses info.description section (#457)
camaraproject/Commonalities: documentation/CAMARA-API-Design-Guide.md

        message: {{Message example}}
```

### 3.3. Error Responses - Mandatory template for `info.description` in CAMARA API specs

The following template must be used as part of the API documentation in the `info.description` property of the CAMARA API specs to provide a common reference for API Consumers, API Developers and API Providers about not documented error responses in case they are supported by a given API implementation.

```md
# Additional CAMARA error responses

The list of error codes in this API specification is not exhaustive. Therefore the API specification may not document some non-mandatory error statuses as indicated in `CAMARA API Design Guidelines`.

Please refer to [CAMARA_common.yaml]({link_to_release}/CAMARA_common.yaml) for a complete list of error responses.

> As an specific rule, error `501 - NOT_IMPLEMENTED` can be only a possible error response if it is explicitly documented in the API.

Reviewer: ```suggestion
As a specific rule, error `501 - NOT_IMPLEMENTED` can be only a possible error response if it is explicitly documented in the API.

```

Reviewer: Just spotted the smallest of typos :)

        message: {{Message example}}
```

### 3.3. Error Responses - Mandatory template for `info.description` in CAMARA API specs

The following template must be used as part of the API documentation in the `info.description` property of the CAMARA API specs to provide a common reference for API Consumers, API Developers and API Providers about not documented error responses in case they are supported by a given API implementation.

```md
# Additional CAMARA error responses

The list of error codes in this API specification is not exhaustive. Therefore the API specification may not document some non-mandatory error statuses as indicated in `CAMARA API Design Guidelines`.

Please refer to [CAMARA_common.yaml]({link_to_release}/CAMARA_common.yaml) for a complete list of error responses.

As a specific rule, error `501 - NOT_IMPLEMENTED` can be only a possible error response if it is explicitly documented in the API.

```

> NOTE: The template MUST be renderized by replacing `{link_to_release}` with the specific value of the Commonalities Release the API specification version is associated to.
> For instance, for an API specification version associated to different Commonalities stages in Spring25 MetaRelease it would have been:
> - API specification version associated to alpha release: `https://github.com/camaraproject/Commonalities/blob/r2.1/artifacts`
> - API specification version associated to release candidate release: `https://github.com/camaraproject/Commonalities/blob/r2.2/artifacts`
> - API specification version associated to public release: `https://github.com/camaraproject/Commonalities/blob/r2.3/artifacts`

Reviewer 2: I would stay with the one example of `{link_to_release}`  only:

```suggestion
NOTE: The template MUST be rendered by replacing `{link_to_release}` with the specific value of the CAMARA Commonalities release the API specification version is compliant to.
For instance: `https://github.com/camaraproject/Commonalities/blob/r3.1/artifacts`

```

Author: Thanks for the suggestion Rafal. It is better

  # title without "API" in it, e.g. "Number Verification"
  title: Number Verification
  # description explaining the API, part of the API documentation 
  # including mandatory text "Authorization and authentication"

> 

Author: ```suggestion
```

  # title without "API" in it, e.g. "Number Verification"
  title: Number Verification
  # description explaining the API, part of the API documentation 

>   # including mandatory texts "Authorization and authentication"

Reviewer: ```suggestion
  # including mandatory text "Authorization and authentication"

```

>   # including mandatory texts "Additional CAMARA error responses"

Reviewer: ```suggestion
  # including mandatory text "Additional CAMARA error responses"

```

  # title without "API" in it, e.g. "Number Verification"
  title: Number Verification
  # description explaining the API, part of the API documentation 

>   # including mandatory texts "Authorization and authentication"

Reviewer: ```suggestion
  # including mandatory text for "Authorization and authentication"

```

>   # including mandatory texts "Additional CAMARA error responses"

Reviewer: ```suggestion
  # including mandatory text for "Additional CAMARA error responses"

```

Title describes the API shortly. The title shall not include the term "API" in it.

#### 5.3.2. Description

> `description`field: There are no special restrictions specified in CAMARA for the documentation explaining API.

Reviewer: ```suggestion
`description`field: There are no special restrictions specified in CAMARA for the documentation explaining APIs.

```
